carl/modules/authsession.py

```diff
@@ -0,0 +1,79 @@
+from flask import current_app
+import requests
+import time
+
+from carl.config import (
+    FHIR_CLIENT_ID,
+    FHIR_CLIENT_SECRET,
+    FHIR_SERVER_AUTH_URL,
+)
+
+
+def _fetch_token():
+    headers = {"Content-Type": "application/json"}
+    body = {
+        "grant_type": "client_credentials",
+        "client_id": FHIR_CLIENT_ID,
+        "client_secret": FHIR_CLIENT_SECRET,
+    }
+    # verify=False supress SSL warnings like curl --insecure  (NOT ADVISED)
+    current_app.logger.debug(f"token request to {FHIR_SERVER_AUTH_URL}")
+    current_app.logger.debug(f"with headers: {headers} and data: {body}")
+    response = requests.get(FHIR_SERVER_AUTH_URL, verify=False, headers=headers, data=body)
```

Comment:
I think this needs to be POST
```suggestion
    response = requests.post(FHIR_SERVER_AUTH_URL, verify=False, headers=headers, data=body)
```

Reply:
spot on!  overlooked the `curl --data` is an implicit POST.  thank you.